cmd/gateway-azure-anonymous.go

```diff
@@ -57,6 +57,28 @@ func (a AzureObjects) AnonGetBucketInfo(bucket string) (bucketInfo BucketInfo, e
 	return bucketInfo, nil
 }
 
+// AnonPutObject - SendPUT request without authentication.
+// This is needed when clients send PUT requests on objects that can be uploaded without auth.
+func (a AzureObjects) AnonPutObject(bucket, object string, size int64, data io.Reader, metadata map[string]string, sha256sum string) (objInfo ObjectInfo, err error) {
+	u := a.client.GetBlobURL(bucket, object)
+	req, err := http.NewRequest("PUT", u, data)
+	if err != nil {
+		return ObjectInfo{}, azureToObjectError(traceError(err), bucket, object)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return ObjectInfo{}, azureToObjectError(traceError(err), bucket, object)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return ObjectInfo{}, azureToObjectError(traceError(anonErrToObjectErr(resp.StatusCode, bucket, object)), bucket, object)
+	}
+
+	return a.AnonGetObjectInfo(bucket, object)
+}
```

Comment:
Azure does not support anonymous PUT, so here you can just return NotImplemented